convert2rhel/unit_tests/actions/conversion/preserve_only_rhel_kernel_test.py

```diff
@@ -0,0 +1,430 @@
+        assert expected.issubset(fix_invalid_grub2_entries_instance.messages)
+
+
+class TestFixDefaultKernel:
+    @pytest.mark.parametrize(
+        ("system_name", "version", "old_kernel", "new_kernel", "not_default_kernels"),
+        (
+            (
+                "Oracle Linux Server release 7.9",
+                Version(7, 9),
+                "kernel-uek",
+                "kernel",
+                ("kernel-uek", "kernel-core"),
+            ),
+            (
+                "Oracle Linux Server release 8.1",
+                Version(8, 1),
+                "kernel-uek",
+                "kernel-core",
+                ("kernel-uek", "kernel"),
+            ),
+            (
+                "CentOS Plus Linux Server release 7.9",
+                Version(7, 9),
+                "kernel-plus",
+                "kernel",
+                ("kernel-plus",),
+            ),
+        ),
+    )
+    def test_fix_default_kernel_converting_success(
+        self,
+        system_name,
+        version,
+        old_kernel,
+        new_kernel,
+        not_default_kernels,
+        caplog,
+        monkeypatch,
+        fix_default_kernel_instance,
+    ):
+        monkeypatch.setattr(system_info, "name", system_name)
+        monkeypatch.setattr(system_info, "arch", "x86_64")
+        monkeypatch.setattr(system_info, "version", version)
+        monkeypatch.setattr(
+            utils,
+            "get_file_content",
+            lambda _: "UPDATEDEFAULT=yes\nDEFAULTKERNEL=%s\n" % old_kernel,
+        )
+        monkeypatch.setattr(utils, "store_content_to_file", StoreContentToFileMocked())
+
+        fix_default_kernel_instance.run()
+
+        warning_msgs = [r for r in caplog.records if r.levelname == "WARNING"]
+        assert warning_msgs
+        assert "Detected leftover boot kernel, changing to RHEL kernel" in warning_msgs[-1].message
+
+        (filename, content), _ = utils.store_content_to_file.call_args
+        kernel_file_lines = content.splitlines()
+
+        assert "/etc/sysconfig/kernel" == filename
+        assert "DEFAULTKERNEL=%s" % new_kernel in kernel_file_lines
+
+        for kernel_name in not_default_kernels:
+            assert "DEFAULTKERNEL=%s" % kernel_name not in kernel_file_lines
+
+    def test_fix_default_kernel_with_no_incorrect_kernel(self, caplog, monkeypatch, fix_default_kernel_instance):
+        monkeypatch.setattr(system_info, "name", "CentOS Plus Linux Server release 7.9")
+        monkeypatch.setattr(system_info, "arch", "x86_64")
+        monkeypatch.setattr(system_info, "version", Version(7, 9))
+        monkeypatch.setattr(
+            utils,
+            "get_file_content",
+            lambda _: "UPDATEDEFAULT=yes\nDEFAULTKERNEL=kernel\n",
+        )
+        monkeypatch.setattr(utils, "store_content_to_file", StoreContentToFileMocked())
+
+        fix_default_kernel_instance.run()
+
+        info_records = [m for m in caplog.records if m.levelname == "INFO"]
```

Comment:
info_records does not seems to be used anywhere in the test

Reply:
We use it in an assert at the bottom of the test